src/components/Localization2DViewer.tsx

```diff
@@ -40,7 +45,75 @@ export default function Localization2DViewer(props: {
       );
       setLinesData(posData);
     };
+
+    const loadNavGraph = async () => {
+      let nav_graph_text = '';
+      const res = await fetch(config.navGraphUrl(props.mapName));
+      if (res.ok) {
+        try {
+          nav_graph_text = await res.text();
+        } catch (err) {
+          console.error(`Bad nav graph data for "${props.mapName}"`, err);
+          return;
+        }
+      }
+
+      interface graphNode {
+        id: number;
+        x: number;
+        y: number;
+        neighbors: Array<number>;
+      }
+
+      let nodes = new Map<number, graphNode>();
+
+      // Collect information about nodes
+      nav_graph_text.split('\n').map((nodeText) => {
+        let node_info = nodeText.split(', ');
+        let neighbors = [];
+        for (
+          let neighbor_idx = 0;
+          neighbor_idx < parseInt(node_info[3], 10);
+          neighbor_idx++
+        ) {
+          neighbors.push(parseInt(node_info[neighbor_idx + 4]));
+        }
+        let node_id = parseInt(node_info[0], 10);
+        nodes.set(node_id, {
+          id: node_id,
+          x: parseFloat(node_info[1]),
+          y: parseFloat(node_info[2]),
+          neighbors: neighbors,
+        } as graphNode);
+      });
+
+      let edges: number[][] = [];
+
+      // Create list of edges with coordinates [x1, y1, x2, y2]
+      for (let node_id of nodes.keys()) {
+        let node = nodes.get(node_id)!;
+        node.neighbors.forEach((neigbhor: number) => {
+          // We only add edges from the smaller neighbor to the larger one, since all our edges are represented twice.
+          if (neigbhor > node_id) {
+            edges.push([
+              node.x,
+              node.y,
+              nodes.get(neigbhor)!.x,
+              nodes.get(neigbhor)!.y,
+            ]);
+          }
+        });
+      }
```

Comment:
what do you think about pulling this out into like a `parseNavGraph(text) -> {nodes, edges}` in another module?